### DISCUSSION TITLE
LOGB separate white point offset?

### DISCUSSION THREAD

**Discussion by madchiller** [gopro/labs#231] [Feature Requests]
I know the LOGB hack is deep, but it would be amazing to have a separate white point offset.    The current offset does not allow for white point to be lowered enough, as it stands, to approximate  a curve such as ASA 200 ARRI LOG  C.     Arri curves can be seen here  https://www.arri.com/resource/blob/31918/66f56e6abb6e5b6553929edf9aa7483e/2017-03-alexa-logc-curve-in-vfx-data.pdf

As EV comp is not available with manual shutter... this would be extremely helpful... This would allow us to use a LOGC gamma CST in Resolve with better accuracy.   Cheers!
Chad

**Comment by dnewman-gpsw** [comment]
When the standard EV is not available you can use oMEVBS=x  where x is any EV comp you need.  This control global, so it is added to any EV-comp currently used in each preset.  Best, it works in modes that don't off EV Comp (livestreaming, and manual shutter.)

It this sufficient to do LOG C emulation?

**Reply by madchiller** [reply]
Thank you for your reply and time.  I will do some testing with this however my gut says no.. I only really mentioned EV comp as I had seen it offered as an idea to lower white point in another post..  I feel that EV comp will additionally alter "mid grey"
 and lower black point and not really get us where we want.. Ideally we would expose "mid grey" appropriately and use the log curve to roll back highlights .  Using base 400 and Black offset around -116 gets us fairly close with a logc gamma transform, but the contrast is much greater..  and highlights are not rolled back quite enough.

**Reply by dnewman-gpsw** [reply]
The white point is controllable by EV, as all sensors are linear, the log and gamma curves are applied later.  The exposure controls are impact the signal, within the linear range of the sensor, form the clipping point to the noise floor.  There is no real black point in the linear space, just the noise floor. The LOGB offset control set where the noise floor appears on the Log curve, again that is cosmetic, as some tools like having a non-zero black level.  LOGB is a pure log curve, not quite the same the third party camera log that mix linear and log segments.

**Reply by madchiller** [reply]
Understand this.. What I was speaking to directly was.. simply being able to roll off highlight via the log curve if all other things are exposed properly..  White point is not clipping, shadows are not crushed into the noise floor... mids are properly exposed..    but highlight roll off still is not enough... The Arri curve as an example accomplishes this via its log curve.. I also under stand the LogB curve implementation doesn't encompass  all the same math .. but if we could some how roll back the highlight via the white  point offset we could get much closer...   If this cannot be accomplished... I understand... and thank you.

**Reply by dnewman-gpsw** [reply]
It is not how log curves work, you don't simply add extra roll-off for highlights.  A log curve means the exposure to code word usage is linear (particularly for the highlights.) Log-C graph here shows the linear response I'm talking about
![image](https://user-images.githubusercontent.com/10856084/217981799-a8ef6207-75b1-4490-ad82-775dbd0b0e16.png)
  
The LOGB control has the same characteristic. Here is LOGB=400
![image](https://user-images.githubusercontent.com/10856084/217984345-e916e92a-3ba3-4d90-88d7-4545170c8231.png)

If you want more highlight protection, but with the same middle grey, than is just math, increase the log base, with decreased EV.   You have all the tools you need, although there are limits in the sensor, the effective DR is way less than an Alexa.
